File: backend/app/schemas/cliente.py
from pydantic import BaseModel, ConfigDict, EmailStr
from datetime import datetime
from typing import Optional

class ClienteBase(BaseModel):
    nombre: str
    nif: Optional[str] = None
    direccion: Optional[str] = None
    ciudad: Optional[str] = None
    codigo_postal: Optional[str] = None
    pais: str = "España"
    email: Optional[EmailStr] = None
    telefono: Optional[str] = None

class ClienteCreate(ClienteBase):
    pass

class ClienteUpdate(BaseModel):
    nombre: Optional[str] = None
    nif: Optional[str] = None
    direccion: Optional[str] = None
    ciudad: Optional[str] = None
    codigo_postal: Optional[str] = None
    pais: Optional[str] = None
    email: Optional[EmailStr] = None
    telefono: Optional[str] = None

class ClienteResponse(ClienteBase):
    id: int
    user_id: str
    created_at: datetime
    updated_at: datetime
    
    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/cliente.ts
// Cliente type definitions based on backend API schemas

export interface ClienteBase {
  nombre: string;
  nif?: string | null;
  direccion?: string | null;
  ciudad?: string | null;
  codigo_postal?: string | null;
  pais?: string;
  email?: string | null;
  telefono?: string | null;
}

export interface ClienteCreate extends ClienteBase {
  pais: string; // Default to "España"
}

export interface ClienteUpdate extends Partial<ClienteBase> {
  // All fields optional for partial updates
}

export interface Cliente extends ClienteBase {
  id: number;
  user_id: string;
  created_at: string; // ISO 8601 datetime string
  updated_at: string; // ISO 8601 datetime string
  pais: string; // Always present in responses
}

export type ClientesListResponse = Cliente[];
export type ClienteResponse = Cliente;